cmd/root.go

```diff
@@ -55,6 +56,33 @@ func RunCommand() *cobra.Command {
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 			output.SetOut(cmd.OutOrStdout())
 			output.SetErr(cmd.ErrOrStderr())
+
+			// Load config from the file (or default if it doesn't exist)
+			configHome, err := getConfigHome(opts)
+			if err != nil {
+				output.Errorf("Failed to read base config directory")
+				output.Infof("Use the --config flag or set the $%s environment variable to provide a default", constants.KitopsHomeEnvVar)
+				output.Debugf("Error: %s", err)
+				return errors.New("exit")
+			}
+
+			configPath := filepath.Join(configHome, constants.ConfigFileName)
+			cfg, err := config.LoadConfig(configPath)
+			if err != nil {
+				if !os.IsNotExist(err) { // If the config file exists but there's an error, report it
+					return output.Fatalf("Failed to load config: %s", err)
+				}
+				cfg = config.DefaultConfig() // Load default config if file doesn't exist
```

Comment:
I think `LoadConfig` should not have a chance of returning an IsNotExist error. Instead, it should catch that error internally and just return DefaultConfig.

Also, note that `os.IsNotExist(err)` is [semi-deprecated](https://pkg.go.dev/os#IsNotExist), and should be replaced with `errors.Is(err, fs.ErrNotExist)`

Reply:
refactored